### ISSUE TITLE
[search] Specific village not found again

### ISSUE THREAD

**Issue by dreng** [organicmaps/organicmaps#7847]
**Describe the issue**
After #3187 has been fixed I thought that this issue has been completely resolved. I tried the same search today and wondered why I couldn't find the village "Wald-Michelbach" again. It turns out that the search result depends on the current map section.

If the current map section is near the searched village it will be found and placed on top of the results, which is as expected. If I move away a bit the village will be found but is **not** placed on top of the results. If I move completely away the village won't be found at all. At first I tought that the village has to be within the current map section. But if I zoom out in order to show the whole of Germany or Europe, the village won't be found either.

**Steps to reproduce**
1. Navigate to "Frankfurt am Main", zoom level approx. 20km
2. Search for "Wald-Michelbach"
3. Notice the correct result at the top
4. Move the map section about 1,5 screens to the north (the city "Paderborn" should just be visible at the top edge)
5. Redo the previous search
6. Find Wald-Michelbach at 4th position now
7. Move the map section complete out of the way, e.g. center at Berlin
8. Redo the search again > Notice that Wald-Michelbach is not part of the search results at all

**Expected behaviour**
Any search with an exact hit should place this hit at the top most position, no matter where you are on the map.

**Screenshots**
Shouldn't be necessary. Please raise your hand if you need some.

**System information:**
 - Android 13
 - 2024.03.31-8-Google
 - Google Pixel 4a

**Comment by vng** [comment]
https://github.com/organicmaps/organicmaps/issues/4670
